src/functionHandlers/getProfile/getProfile.ts

```diff
@@ -18,9 +18,22 @@ const getIdentifier = async (event: APIGatewayEvent) => {
   if (!id) {
     throw new createHttpError.BadRequest("Identifier is not provided");
   }
+
+  const apiKey =
+    event.headers["x-api-key"] ||
+    event.headers.Authorization ||
+    event.headers["api-key"] ||
+    event.headers.apikey ||
+    config.sheetsApiKey;
```

Comment:
The server will have it's own API key location, choose one. The client will have to follow. You just have to solve the problem of not know what the server is expecting in a generic web app, so on the web app you will define both:

header: api-key
key: abc-123